openedx/frontend-app-course-authoring
# feat: proctoring & prerequisite settings and page alerts

**Description:**

This PR adds following features to course outline:

* Proctoring exam link action in subsection menu.
* Adds settings related to prerequisites to advanced tab of subsection configuration.
* Adds settings related to proctoring to advanced tab of subsection configuration.
* Adds page alerts to course outline (screenshot can be found below).
* Adds discussion badge to units.
* Also updates UI to reflect latest changes in figma design.

**Test instructions:**

* Get devstack up and running using `make {lms,cms}-up`.
* Start npm devserver in this repo using `npm start`.
* Open [edx-platform/cms/djangoapps/contentstore/utils.py](https://github.com/open-craft/edx-platform/blob/4a75d9845d7c0ccb81bc0934eec92156b08f968b/cms/djangoapps/contentstore/utils.py#L1760-L1760) in your editor and modify `discussion_settings` or `mfe_proctored_exam_settings_url` or `deprecated_blocks_info` etc. to see alerts in MFE.
* Go to http://localhost:2001/course/course-v1:edX+DemoX+Demo_Course/
* Play around with section, subsection & unit configuration.

<details>
  <summary>Screenshots:</summary>
 
![Screenshot 2024-02-02 at 15-40-59 Course outline Demonstration Course Your Plaform Name Here](https://github.com/openedx/frontend-app-course-authoring/assets/10894099/437b8c9a-aff8-4b3a-9b18-54480fe68fff)

![image](https://github.com/openedx/frontend-app-course-authoring/assets/10894099/e5818545-a558-464b-bae5-f326b38b8dbd)

![image](https://github.com/openedx/frontend-app-course-authoring/assets/10894099/84917cf2-0cd9-44d6-b01c-b3ae232bf083)

![image](https://github.com/openedx/frontend-app-course-authoring/assets/10894099/212a9b2d-64c0-4c2d-80c4-14ed5317a89a)




</details>
